x-pack/plugins/observability_solution/entity_manager/server/routes/enablement/enable.ts

```diff
@@ -79,20 +74,19 @@ export function enableEntityDiscoveryRoute<T extends RequestHandlerContext>({
         const apiKey = await generateEntityDiscoveryAPIKey(server, req);
 
         if (apiKey === undefined) {
-          throw new Error('could not generate entity discovery API key');
+          return res.customError({
```

Comment:
what's the reason for this change vs letting the `catch` at the end handle it?

Reply:
personal preference, why throw when we can return since we're already in the route